doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,941 @@
+
+The specialized DPUs when built into a regular switch, can provide such a capability, which is being referred as SmartSwitch.
+Platform monitor PMON in SONiC is a container responsible for chassis management functions to ensure proper operation of the devices and peripherals in the chassis there by ensuring the proper operation of the product. 
+
+The typical lifecycle of a product involves the following stages.
+
+<p align="center"><img src="./images/lifecycle.svg"></p>
+
+The following sub-tasks are performed under each stage
+* Onboarding
+    * Boot, Shutdown, Power Cycle
+    * Rest, PCIe-Reset
+* Monitoring
+    * Device State (dpu_state)
+    * Sensors, PSUs, Cooling Devices, Thermal management
+    * Show CLIs
+* Detection and Debugging
+    * DPU Health
+    * Alarms, Syslog
+    * Console
+* RMA
+    * Inventory
+
+The purpose of this document is to provide a framework to share the state, health, alarms of the DPUs, manage the DPUs by providing support to monitor, gracefully shutdown, restart them and the associated peripherals such as thermal sensors, cooling devices, LEDs, etc.
+
+The picture below highlights the PMON vertical and its association with other logics within the SONiC architecture.
+
+<p align="center"><img src="./images/pmon-vertical.svg"></p>
+
+## 2.	Requirements and Assumptions
+
+### 2.1.    Onboarding
+* The SmartSwitch host PMON should be able to Power Cycle, Shutdown, Reset, and reset the PCIe link per DPU or the entire system. The DPU_MODULE will behave like the LINE_CARD_MODULE of a modular chassis with respect to these functions.
+
+#### SmartSwitch PowerUp sequence:
+* When the smartswitch device is booted, the host will boot first and leave the DPUs in powered up state by default.
+* The DPUs can be powered down by configuring the admin_status as shown.
+* The corresponding switch stateDB table is also shown
+
+#### config_db.json
+CHASSIS_MODULE table holds the list and configuration of DPU modules in a smartswitch chassis. It allows user to administratively bring down a DPU
+```
+{
+    "CHASSIS_MODULE": {
+        "DPU0": {
+            "admin_status": "up"
+        },
+        "DPU1": {
+            "admin_status": "up"
+        }
+    }
+}
+```
+#### switch stateDB
+```
+    Key: "CHASSIS_MODULE_TABLE|DPU0"
+    "CHASSIS_MODULE_TABLE|DPU0":{
+        "value": {
+            "admin_status":"up",
+        }
+    }
+```
+#### Chassis/DPU power on sequence
+* The chassis is powered up and the host is booting up.
+* PMON gets DB notification and invokes ModuleClass API set_admin_state(self, up)
+* The host powers up the DPU (TBD: GNOI). NPU-DPU (GNOI) reboot workflow will be captured in another document
+* DPU boots up and attaches itself to the midplane.
+* Once SONiC is up the state progression is updated for every state transition on the DPU_STATE table in the chassisStateDB
```

Comment:
These are all fine, but like to iterate that the minimal set of DPU_STATE will be updated into the ChassisStateDB by the DPUs as state changes.  This was requested primarily by the HA team as well.  Are you saying that this data should be populated by the DPUs into the StateDB instead of the ChassisStateDB?  If this is what you are requesting please let us knwo.